converter.go

```
@@ -3219,6 +3200,25 @@ func convertTzTypeToSnowflakeType(tzType timezoneType) snowflakeType {
 	return unSupportedType
 }
 
+func getTimestampBindValue(x time.Time, stream bool, t snowflakeType) string {
+	if stream {
+		return x.Format(format)
+	}
+	return convertTimeToTimeStamp(x, t)
+}
+
+func convertTimeToTimeStamp(x time.Time, t snowflakeType) string {
+	unixTime, _ := new(big.Int).SetString(fmt.Sprintf("%d", x.Unix()), 10)
+	m, _ := new(big.Int).SetString(strconv.FormatInt(1e9, 10), 10)
```

Maybe we should propagate the errors here?

---

MO, error handling is not necessary here because the FormatInt function cannot fail—it operates on a constant (essentially hardcoded). This means the m value is always 1,000,000,000, which is why we do not handle the error in this case.
https://github.com/snowflakedb/gosnowflake/pull/1297/files#diff-ecad48054a05ca5aecf42b278530331772747ed7211a193aacd94214cb53aad0L906

---

That is true for today. But you never know when someone changes the other behavior that may influence this one, right? It is safer to always handle errors, even if today's code never fails.

---

Make sense. I added error handling.

---

Yes, I know, this is a problem with Go coverage :) In other languages this is just exception, no extra lines for propagating error up.